pulp_rpm/app/tasks/synchronizing.py

```diff
@@ -306,9 +309,22 @@ def newpkgcb(pkgId, name, arch):
                     updateinfo_url = urljoin(remote_url, record.location_href)
                     downloader = self.remote.get_downloader(url=updateinfo_url)
                     downloaders.append([downloader.run()])
-                else:
-                    log.info(_('Unknown repodata type: {t}. Skipped.').format(t=record.type))
-                    # TODO: skip databases, save unknown types to publish them as-is
+                elif record.type not in PACKAGE_DB_REPODATA:
+                    file_data = {record.checksum_type: record.checksum, "size": record.size}
+                    da = DeclarativeArtifact(
+                        artifact=Artifact(**file_data),
+                        url=urljoin(remote_url, record.location_href),
+                        relative_path=record.location_href,
+                        remote=self.remote,
+                        deferred_download=self.deferred_download
```

Comment:
Historically we had all the repo metadata always downloaded.
I'm not sure if it's a good or bad idea to download it on_demand only. 

One example (in favor of always downloading repository metadata):
 - sync a repo with on_demand policy
 - copy custom metadata to another repo
 - in the meantime metadata changes upstream, previous metadata is no longer available to be downloaded.

To be fair, it can happen to any content, however metadata is a bit different, it's always relevant to a specific set of content in a repo, so when you copy it, you do want to preserve the "old" metadata because it will be relevant to the content you have in your copy.

I think my suggestion would be to always set `deferred_download` to `False`.

Does it make sense?
Any thoughts? @fabricio-aguiar @daviddavis

Reply:
@daviddavis , @dkliban , @ipanova , what do you think about ALWAYS downloading custom repository metadata?

Reply:
+1 to always download custom repo metadata

Reply:
let's download it all the time